src/utils/catchError.js

```
@@ -0,0 +1,9 @@
+export const catchError = (action) => {
+  return async function (req, res, next) {
+    try {
+      await action.call(this, req, res, next);
```

You're calling `await action.call(this, req, res, next);` which preserves context. As a small defensive improvement, you could validate that `action` is a function before calling it to fail early with a clearer error if misused (e.g., `if (typeof action !== 'function') throw new Error('catchError expects a function')`). This is optional but helpful during debugging.